### DISCUSSION TITLE
settings.tgz.crypted entschlüsseln klappt nicht "-md" fehlt

### DISCUSSION THREAD

**Discussion by Conan179** [Freetz-NG/freetz-ng#1073] [Allgemein]
```
Hi ich wollte meine settings.tgz.crypted entschlüsseln, wie es im contents.txt drin steht, aber der befehl:
openssl enc -d -aes256 -in settings.tgz.crypted -out settings.tgz [-pbkdf2| -m
d sha256]
Spukt nur die Fehlermeldung:
enc: Use -help for summary.
Command '-md' not found, did you mean:
  command 'wmd' from deb wml (2.32.0~ds1-1)
  command 'umd' from deb node-umd (3.0.3+ds+~3.0.1-1)
  command 'emd' from deb emd (1.0.1-3ubuntu0.24.04.1)
  command 'mmd' from deb mtools (4.0.43-1)
Try: sudo apt install <deb name>
aus.
Getestet auf 2 verschiedenen Systemen.
```

**Comment by fda77** [comment]
nimm die gleich openssl-hauptversion wie auf der box. manches fällt in neuen versionen weg

**Reply by Conan179** [reply]
Also auf der Box läuft 
OpenSSL 3.0.15 3 Sep 2024 (Library: OpenSSL 3.0.15 3 Sep 2024),
 auf meiner Debian innstallation läuft 
OpenSSL 3.0.14 4 Jun 2024 (Library: OpenSSL 3.0.14 4 Jun 2024) 
und auf WSL 
OpenSSL 3.0.13 30 Jan 2024 (Library: OpenSSL 3.0.13 30 Jan 2024)

Die hauptversion müste 3.0.x sein, die ist auf allen drauf, das auf der box ist sogar neuer wie bei den system wo ich das encrypten getestet habe.

**Comment by fda77** [comment]
Die Versionen passen. Weil du den Code nicht als Code formatiert hast sieht man es nicht so gut .. hast du wirklich die Klammern mit eingegeben?

**Reply by Conan179** [reply]
Ich hab copy paste, direkt aus der Text datei gemacht, mit editpadpro wo mit Linux zeilen umbrüchen umgehen kann.

**Comment by fda77** [comment]
du musst dir halt den parameter aussuchen der für dich passt, bzw den den dein openssl kann

**Reply by Conan179** [reply]
achso [-pbkdf2|-md sha256] bedeutet entweder oder... 
openssl enc -d -aes256 -in settings.tgz.crypted -out settings.tgz -md sha256
kann es mit passwort nicht entschlüsseln
openssl enc -d -aes256 -in settings.tgz.crypted -out settings.tgz -pbkdf2
kann es mit password entpacken, sorry das war für mich nicht ersichtlich, das es entweder oder heisen soll. Tschuldigung

**Comment by fda77** [comment]
Wenn das so wie es ist schwer zu verstehen ist könnte man es auch in 2 Zeilen schreiben.
Aber eigentlich ist das Linux Grundwissen und wird überall so genutzt, zb
```
$ rc.dnsmasq help
Usage: /mod/etc/init.d/rc.dnsmasq [load|unload|start|stop|restart|reload|status]
```
```
$ busybox --help
Usage: busybox [function [arguments]...]
```

```
$ sort --help
Usage: sort [-nrughMcszbdfiokt] [-o FILE] [-k START[.OFS][OPTS][,END[.OFS][OPTS]] [-t CHAR] [FILE]...
```

**Reply by Conan179** [reply]
Ich muss mich endschuldigen, ich hatte es das nicht erkannt -.-
